DashAI/back/dataloaders/classes/dataloader.py

```diff
@@ -0,0 +1,70 @@
+from abc import abstractmethod
+
+from datasets import ClassLabel
+
+from DashAI.back.config_object import ConfigObject
+
+
+class DataLoader(ConfigObject):
+    """
+    Abstract class with base methods for all data loaders
+    """
+
+    @abstractmethod
+    def load_data(self, dataset_path, url=None):
+        pass
+
+    @abstractmethod
+    def set_task_format(self, task):
+        pass
+
+    def split_dataset(self, dataset, params, class_column):
+        """
+        Split the dataset in train, test and validation data
+        """
```

Comment:
You're right is necessary a good documentation, I'll leave all the documentation in this branch tomorrow. Now in the branch front/dataloaders are more documentation.

Answering the questions:
1- The parameters are passed in this dictionary because in the endpoint this is obtained directly from the json received, so the 'splits' key is passed directly. But it effectively doesn't make sense to have this and separate the class column. I agree that it is better to make the parameters for this method explicit, I will change it.

2- From what was discussed in meetings I assumed that the three partitions had to exist, but it was thought that I could choose not to have a split by giving it a size of 0. Now that I see it better, this is not working with a partition at 0, but I can correct it to let him do it.

3- If the splits are defined before, this method is not called. This depend of a parameter ('folder_split' ir now renamed as 'splits_in_folders') that is checked in the endpoint.

Reply:
Thanks for the answers!

Regarding point 2, if you think that the behavior when giving 0 in some split is wrong, then at the beginning verify that the parameter is greater than 0 and in the case that it is not, raise an exception. 

```python
if not isinstance(val_size, float):
    raise TypeError("val_size muest be float, got {type(val_size)}") .

if val_size < 0 or val_size > 1:
    raise ValueError("val_size must be in the (0, 1) range (0 and 1 not included), got {val_size}")
```

Reply:
Thanks for your advice! I understand, I will clarify it in the code.